src/joystick/hidapi/SDL_hidapi_sinput.c

```diff
@@ -0,0 +1,798 @@
+
+                if ( (output_idx < SDL_GAMEPAD_BUTTON_COUNT) && (ctx->last_state[button_idx] != data[button_idx]) ) {
+                    SDL_SendJoystickButton(timestamp, joystick, output_idx, down);
+                }
+
+                ++output_idx;
+            }
+        }
+    }
+
+    // Analog inputs map to a signed Sint16 range of -32768 to 32767 from the device.
+    // Use an axis index because not all gamepads will have the same axis inputs.
+    Uint8 axis_idx = 0;
+
+    // Left Analog Stick
+    axis = 0; // Reset axis value for joystick
+    if (ctx->left_analog_stick_supported) {
+        axis = EXTRACTSINT16(data, SINPUT_REPORT_IDX_LEFT_X);
+        SDL_SendJoystickAxis(timestamp, joystick, axis_idx, axis);
+        ++axis_idx;
+
+        axis = EXTRACTSINT16(data, SINPUT_REPORT_IDX_LEFT_Y);
+        SDL_SendJoystickAxis(timestamp, joystick, axis_idx, axis);
+        ++axis_idx;
+    }
+    
+    // Right Analog Stick
+    axis = 0; // Reset axis value for joystick
+    if (ctx->right_analog_stick_supported) {
+        axis = EXTRACTSINT16(data, SINPUT_REPORT_IDX_RIGHT_X);
+        SDL_SendJoystickAxis(timestamp, joystick, axis_idx, axis);
+        ++axis_idx;
+
+        axis = EXTRACTSINT16(data, SINPUT_REPORT_IDX_RIGHT_Y);
+        SDL_SendJoystickAxis(timestamp, joystick, axis_idx, axis);
+        ++axis_idx;
+    }
+
+    // Left Analog Trigger
+    axis = SDL_MIN_SINT16; // Reset axis value for trigger
+    if (ctx->left_analog_trigger_supported) {
+        axis = EXTRACTSINT16(data, SINPUT_REPORT_IDX_LEFT_TRIGGER);
+    }
+
+    // Our digital trigger L will override the analog value
+    if (ctx->digital_trigger_l) {
+        axis = SDL_MAX_SINT16;
+    }
+
+    if (ctx->left_analog_trigger_supported || ctx->left_digital_trigger_supported) {
+        SDL_SendJoystickAxis(timestamp, joystick, axis_idx, axis);
+        ++axis_idx;
+    }
+    
+    
+    // Right Analog Trigger
+    axis = SDL_MIN_SINT16; // Reset axis value for trigger
+    if (ctx->right_analog_trigger_supported) {
+        axis = EXTRACTSINT16(data, SINPUT_REPORT_IDX_RIGHT_TRIGGER);
+    }
+    // Our digital trigger L will override the analog value
+    if (ctx->digital_trigger_r) {
+        axis = SDL_MAX_SINT16;
+    }
+
+    if (ctx->right_analog_trigger_supported || ctx->right_digital_trigger_supported) {
+        SDL_SendJoystickAxis(timestamp, joystick, axis_idx, axis);
+    }
+
+    // Battery/Power state handling
+    if (ctx->last_state[SINPUT_REPORT_IDX_PLUG_STATUS]  != data[SINPUT_REPORT_IDX_PLUG_STATUS] ||
+        ctx->last_state[SINPUT_REPORT_IDX_CHARGE_LEVEL] != data[SINPUT_REPORT_IDX_CHARGE_LEVEL]) {
+
+        SDL_PowerState state;
+        Uint8 status = data[SINPUT_REPORT_IDX_PLUG_STATUS];
+        int percent = data[SINPUT_REPORT_IDX_CHARGE_LEVEL];
+
+        percent = SDL_clamp(percent, 0, 100); // Ensure percent is within valid range
+
+        switch (status) {
+        case 0:
```

Comment:
Protocol suggestion, have 0 be a "status not available" value so it can be left blank for hardware that doesn't report power status.